### PR TITLE
[16.0] [MIG] document_hosting

### PR DESCRIPTION
**Pull request by victor-champonnois**
## Description

- Migrating to 16
- fix settings view and put it in it's own section
- add roadmap

## Odoo task (if applicable)

https://gestion.coopiteasy.be/web#id=11971&action=475&active_id=492&model=project.task&view_type=form&menu_id=

## Checklist before approval

- [ ] Tests are present (or not needed).
- [ ] Credits/copyright have been changed correctly.
- [ ] Change log snippet is present.
- [ ] (If a new module) Moving this to OCA has been considered.

### CODE REVIEW

**Comment by polchampion** [comment]
@victor-champonnois Can you put it in test?

**Comment by victor-champonnois** [comment]
@polchampion done

**Review by robinkeunen — APPROVED** [review]
LGTM

**Review by polchampion — CHANGES_REQUESTED** [review]
functional test on [16-test-divers](https://odoo16.main.test.srv.coopiteasy.be/web?db=16-test-divers)

- [x] internal user can create, upload, share and make documents public
- [ ] public user can display and download public documents **NOK : see, but cannot open nor download**
- [ ] portal user can  display and download published documents **NOK, only sees the public document - same error when opening or downloading**

500: Internal Server Error

**Comment by victor-champonnois** [comment]
@polchampion thanks for the test, it's fixed. I've also fixed the display of category tree (migrating to bootstrap v5)

**Comment by victor-champonnois** [comment]
@polchampion ...and it's just deployed to test

**Comment by polchampion** [comment]
functional test on [16-test-divers](https://odoo16.main.test.srv.coopiteasy.be/web?db=16-test-divers)

- [x] internal user can create, upload, share and make documents public
- [x] public user can display and download public documents
- [ ] portal user can  display and download published documents **NOK, only sees the public document**

@victor-champonnois Only public documents are displayed in Documents, whereas all published documents should show up for portal and internal users, right?

**Comment by victor-champonnois** [comment]
@polchampion to me it's working. As a portal user I see the doc "test partage doc interne" that is published but not public. 
It's in the "categorie document de test though", so you need to click on the category to see it. 
image

**Comment by polchampion** [comment]
functional test on [16-test-divers](https://odoo16.main.test.srv.coopiteasy.be/web?db=16-test-divers)

- [x] internal user can create, upload, share and make documents public
- [x] public user can display and download public documents
- [x] portal user can  display and download published documents

@victor-champonnois Oh I had totally missed that, thank you. I'll suggest a change to better display public documents as a category, but as far as the porting goes, it all looks good to me

**Review by polchampion — APPROVED** [review]

File: document_hosting/controllers/main.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+            request.env["ir.binary"]
+            ._get_image_stream_from(
+                doc, field_name="document", filename_field="filename"
+            )
+            .get_response()
+        )
+
+    @http.route("/documents", auth="public", website=True)
+    def template_website_document(self, date_begin=None, date_end=None, **kw):
+        """"""
```
Comment (remytms) [inline document_hosting/controllers/main.py]: Remove empty docstring or privide a usefull one.

**Review by remytms — APPROVED** [review]
Cosmetic change requested. But it's not mandatory.

File: document_hosting/controllers/main.py (hunk 1 above)
Comment (victor-champonnois) [reply document_hosting/controllers/main.py]: Fixed ! thanks for the review

**Pull request merged**